File: 백준/Silver/2579. 계단 오르기/계단 오르기.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        int n = s.nextInt(); 
        int stair[]=new int[n];
        for (int i = 0; i < n; i++) {
            stair[i] = s.nextInt();
        }
        int[] dp = new int[n];
        for (int i = 0; i < n; i++) { 
            if(i==0) {
            	dp[0]=stair[0];
            } else if (i==1) {
            	dp[1] = Math.max(stair[0] + stair[1], stair[1]);
            } else if (i==2) {
            	dp[2] = Math.max(stair[1] + stair[2], dp[0] + stair[2]);
            } else {
            	dp[i] = Math.max(dp[i - 3] + stair[i - 1] + stair[i], dp[i - 2] + stair[i]);
            }
        }
        System.out.println(dp[n-1]);
    }
}

File: 백준/Silver/2579. 계단 오르기/계단 오르기.py
n = int(input())
stair = [int(input()) for _ in range(n)]
dp = [0] * len(stair)

for i in range(n):
    if i == 0:
        dp[0] = stair[0]
    elif i == 1:
        dp[i] = max(stair[i-1] + stair[i], stair[i])
    elif i == 2:
        dp[i] = max(stair[i-1] + stair[i], dp[i-2] + stair[i])
    else:
        dp[i] = max(dp[i-3] + stair[i-1] + stair[i], dp[i-2] + stair[i])

print(dp[-1])